core/src/config.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as controls from './controls';
+import * as data from './data';
+
+// TODO(iansimon): support multiple control signals
+
+export interface ConfigSpec {
+  dataConverter: data.ConverterSpec;
```

Comment:
I think we also need a section for model args (like attention length).

I'm wondering if we should actually call this ModelSpec and have it include the model type (for now, MusicRNN | MusicVAE) as well as model args. Then we can just have a modelFromCheckpoint function. We would move the loading of checkpoints out of the model classes and make the vars be an argument.

I tmay also be good to move all of the data and control specs into here, but that's not as important.

Reply:
Are you imagining the modelFromCheckpoint function returns an object of type MusicRNN | MusicVAE?  I'm not sure how I feel about that; I think there should be one function (possibly just new()) to instantiate one and one to instantiate the other, since the interfaces for using the models are so different.

What else is there (besides attention length) that isn't part of a DataConverter or ControlSignal?  Maybe attention should actually be a ControlSignal...